cmd/apidemic/main.go

```diff
@@ -1,18 +1,30 @@
 package main
 
 import (
+	"bytes"
+	"encoding/json"
+	"errors"
 	"fmt"
+	"io/ioutil"
 	"log"
 	"net/http"
+	"net/http/httptest"
 
 	"github.com/codegangsta/cli"
 	"github.com/gernest/apidemic"
+	"github.com/gorilla/mux"
 )
 
 func server(ctx *cli.Context) {
 	port := ctx.Int("port")
+	endpointDir := ctx.String("endpoint-dir")
 	s := apidemic.NewServer()
 
+	err := addEndpoints(s, endpointDir)
```

Comment:
The endpoint dir should be optional. Not everyone wants to load stuff from a dir. I think the workflow should be like this.

 - If the user provided a path to the endoints then use it to lad endpoint and exit if any error occured.

- If the use didn't provide an endpoint , try to check if `endpoints/` is there if not continue execution ignore the loading . In case the dir is there ☝️  see above bullet but use `endpoints/` instead.

- Else just ignore the loading

Reply:
:ok: